### DISCUSSION TITLE
How to check if a monitor is connected?

### DISCUSSION THREAD

**Discussion by Jasper-Dev** [WarmUpTill/SceneSwitcher#747] [General]
I can't for the life of me figure out the operators to accomplish my 'seemingly easy usecase'.

Im using OBS on my laptop (1920x1080) and have two external monitors (both 1920x1080) connected to it, so in total I use 3 screens (3 x 1920x1080). Per screen I have set-up a dedicated Scène with all the needed sources. I use AdvancedSceneSwitcher to automatically switch to the screen the my cursor is currently on.

The automatically switching between the scenes works great, and was relatively easy to set-up. 

The issue I'm having is when I disconnect my external monitors and walk away with the laptop.

I'm using the `if Cursor is in region x... y...` to determine which scène needs to be switched to based on cursor location. When I disconnect the external monitors the coordinations are then "reset" to only the laptop screen which happen to be the yx coords of one of the monitors when i have them connected. This results in OBS constantly switching to the external monitor Scène when actually want it to switch to my laptop screen Scène.

I've had multiple tried at it, but I just can't figure it out which option to choose and how to properly set it up. I tried the following things for each of the 4 macro's and their corresponding Scène (so 'External Monitor 1' here is a placeholder) (and yes, it's quite a tedious work):
- `IF NOT Video Source 'External Monitor 1' has no output //this is the closest i've come, but when the corresponding scene is not active the source has no output...`
- `IF NOT Source 'External Monitor 1' Settings Match {...} //this didn't work as the Sources don't switch monitor id...`
- `All sources on 1 Scene and Toggling between the visibility of the sources (scene items) // doesnt really work and adds a whole lot of unnecessary complexity`
- `IF Video Source 'External Monitor 1' does not match {screenshot} //doesnt work`
- `IF NOT Source 'External Monitor 1' is active //doesnt work, because when the Scène isnt active the source isnt either`

when I say 'doesnt work', I mean it doesnt result in the behaviour I want it to behave'.

I'm looking for advice on how to handle this. Perhaps there's a way to create a script or tool that detects monitor connection changes and adjusts the AdvancedSceneSwitcher settings? Or maybe there's a simpler solution, like different scene switcher profiles for connected and disconnected monitors?

Any advice or input from those who have faced similar issues would be greatly appreciated. Thank you!

TL;DR: I'm using OBS with AdvancedSceneSwitcher for a three-monitor setup, each with its own scene. It works great until I disconnect my external monitors. The cursor's coordinates are then limited to the laptop screen and match one of the disconnected monitors, causing OBS to constantly switch to the external monitor scene instead of staying on the laptop scene. I'm seeking advice on a solution to set up the AdvancedSceneSwitcher in a way to detect if when monitors are disconnected.

**Comment by WarmUpTill** [comment]
I can't really think of a solution with the existing functionality available in the plugin.
I will investigate if it is feasible to add some condition type to check the amount of connected monitors tomorrow.

**Comment by WarmUpTill** [comment]
I now implemented the "Display" condition type which should enable you to achieve what you are trying to do.

![Dispaly](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/e8295503-5014-49f1-88d7-5fabbb19d58a)

A build will be available here in a few minutes:
https://github.com/WarmUpTill/SceneSwitcher/actions/runs/4969193920?pr=749

Let me know if you have questions or run into any issues! :)

**Reply by Jasper-Dev** [reply]
Woah, thanks for the quick response and solution for the issue! I've just tested and played around with it for a bit, and my first impression is that it looks really promising! 

I do however, have a minor suggestion regarding the new feature. Rather than specifying each screen individually with: `if Display name of connected display matches ...`, I think it would be easier to use, if there is the ability to select an array of screens as a set or a group. The condition could be something like: `if these displays are connected simultaneously ...`  rather than having to specify `if display name of connected display matches 'display 1' AND display name of connected display matches 'display 2'`. I believe this change would also make the 'NOT' operator variant significantly more easy to use, since a laptop screen doesn't really disconnect, making it harder to distinguish between '2 external monitors' and 'a laptop with 1 external monitor'.

Again, it's just some food for thought, i'm already very happy that the functionality is being implemented at all, so huge thanks for that! :D

**Reply by WarmUpTill** [reply]
Matching multiple screen names at the same time is actually already possible using "regular expressions". I just cropped the option to enable them out of the screenshot, just in case it would be confusing.
Instead of writing just one screen name one would write something like this instead:
`(some display name|some other display name)`
